FTBFS -Werror=array-bounds

I'm trying to build cockroachdb but compilation fails because of some C++ error.

Probably you are running an old gcc version so you don't get those checks.


```
gcc (Debian 9.2.1-8) 9.2.1 20190909
```

```
[  2%] Building CXX object CMakeFiles/roach.dir/protos/roachpb/data.pb.cc.o
/usr/bin/c++   -I/home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/./include -I/home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/../protobuf/src -I/home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/../rocksdb/include -I/home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/protos  -g1 -O3 -DNDEBUG   -Werror -Wall -Wno-sign-compare -std=c++11 -o CMakeFiles/roach.dir/protos/roachpb/data.pb.cc.o -c /home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/protos/roachpb/data.pb.cc
In member function 'void cockroach::roachpb::StoreIdent::SharedCtor()',
    inlined from 'cockroach::roachpb::StoreIdent::StoreIdent()' at /home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/protos/roachpb/data.pb.cc:1179:13,
    inlined from 'void protobuf_roachpb_2fdata_2eproto::InitDefaultsStoreIdent()' at /home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/protos/roachpb/data.pb.cc:1175:1:
/home/salvo/go/src/github.com/cockroachdb/cockroach/c-deps/libroach/protos/roachpb/data.pb.cc:1198:11: error: 'void* memset(void*, int, size_t)' offset [29, 32] from the object at 'cockroach::roachpb::_StoreIdent_default_instance_' is out of the bounds of referenced subobject 'cockroach::roachpb::StoreIdent::node_id_' with type 'int' at offset 24 [-Werror=array-bounds]
 1198 |   ::memset(&node_id_, 0, static_cast<size_t>(
      |   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1199 |       reinterpret_cast<char*>(&store_id_) -
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1200 |       reinterpret_cast<char*>(&node_id_)) + sizeof(store_id_));
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Environment:**
 - CockroachDB version git master
 - debian bullseye

**Additional context**

I can't use it since I can't build it.

--- a/c-deps/libroach-rebuild
+++ b/c-deps/libroach-rebuild
@@ -1,4 +1,4 @@
 Bump the version below when changing libroach CMake flags. Search for "BUILD
 ARTIFACT CACHING" in build/common.mk for rationale.
 
-3
+4

--- a/c-deps/libroach/CMakeLists.txt
+++ b/c-deps/libroach/CMakeLists.txt
@@ -90,7 +90,7 @@ set_target_properties(roach roachccl PROPERTIES
   CXX_STANDARD 11
   CXX_STANDARD_REQUIRED YES
   CXX_EXTENSIONS NO
-  COMPILE_OPTIONS "-Werror;-Wall;-Wno-sign-compare"
+  COMPILE_OPTIONS "-Werror;-Wall;-Wno-sign-compare;-Wno-array-bounds"
 )
 
 enable_testing()
@@ -178,7 +178,7 @@ foreach(tsrc ${tests})
     CXX_STANDARD 11
     CXX_STANDARD_REQUIRED YES
     CXX_EXTENSIONS NO
-    COMPILE_OPTIONS "-Werror;-Wall;-Wno-sign-compare"
+    COMPILE_OPTIONS "-Werror;-Wall;-Wno-sign-compare;-Wno-array-bounds"
   )
 
   # Add the executable to the set of tests run by the "check" target.

